src/server/game/AuctionHouse/AuctionHouseMgr.cpp

```diff
@@ -389,6 +400,119 @@ bool AuctionHouseMgr::RemoveAItem(uint32 id, bool deleteItem)
     return true;
 }
 
+void AuctionHouseMgr::PendingAuctionAdd(Player* player, AuctionEntry* aEntry)
+{
+    PlayerAuctions* thisAH;
+    auto itr = pendingAuctionMap.find(player->GetGUID());
+    if (itr != pendingAuctionMap.end())
+        thisAH = itr->second.first;
+    else
+    {
+        thisAH = new PlayerAuctions;
+        pendingAuctionMap[player->GetGUID()] = AuctionPair(thisAH, 0);
+    }
+    thisAH->push_back(aEntry);
+}
+
+uint32 AuctionHouseMgr::PendingAuctionCount(const Player* player) const
+{
+    auto const itr = pendingAuctionMap.find(player->GetGUID());
+    if (itr != pendingAuctionMap.end())
+        return itr->second.first->size();
+
+    return 0;
+}
+
+void AuctionHouseMgr::PendingAuctionProcess(Player* player)
+{
+    auto iterMap = pendingAuctionMap.find(player->GetGUID());
+    if (iterMap == pendingAuctionMap.end())
+        return;
+
+    PlayerAuctions* thisAH = iterMap->second.first;
+
+    SQLTransaction trans = CharacterDatabase.BeginTransaction();
+
+    uint32 totalItems = 0;
+    for (auto itrAH = thisAH->begin(); itrAH != thisAH->end(); ++itrAH)
+    {
+        AuctionEntry* AH = (*itrAH);
+        totalItems += AH->itemCount;
+    }
+
+    uint32 totaldeposit = 0;
+    auto itr = (*thisAH->begin());
+
+    if (Item* item = GetAItem(itr->itemGUIDLow))
+         totaldeposit = GetAuctionDeposit(itr->auctionHouseEntry, itr->etime, item, totalItems);
+
+    uint32 depositremain = totaldeposit;
+    for (auto itr = thisAH->begin(); itr != thisAH->end(); ++itr)
+    {
+        AuctionEntry* AH = (*itr);
+
+        if (next(itr) == thisAH->end())
+            AH->deposit = depositremain;
+        else
+        {
+            AH->deposit = totaldeposit / thisAH->size();
+            depositremain -= AH->deposit;
+        }
+
+        AH->DeleteFromDB(trans);
+        AH->SaveToDB(trans);
+    }
+
+    CharacterDatabase.CommitTransaction(trans);
+    pendingAuctionMap.erase(player->GetGUID());
+    delete thisAH;
+    player->ModifyMoney(-int32(totaldeposit));
+}
+
+void AuctionHouseMgr::UpdatePendingAuctions()
+{
+    if (pendingAuctionMap.size() == 0)
```

Comment:
this check is not needed (redundant) since it the for won't do any loop if the map is empty